analysis/quantifier-comparisons/15-demux-comparisons.Rmd

```diff
@@ -0,0 +1,290 @@
+demux_tsv <- file.path(outdir, paste0(params$lib_id, "_demux.tsv"))
+demux_tables <- file.path(outdir, paste0(params$lib_id, "_demux_table.txt"))
+
+plotdir <- "plots"
+demux_umap_plot <- file.path(plotdir, paste0(params$lib_id, "_demux_umap.png"))
+```
+
+## Functions
+
+```{r}
+# calculate the agreement between two lists, with and or without NAs included
+calc_agreement <- function(call_a, call_b, na.include = FALSE){
+  agree <- sum(call_a == call_b, na.rm = TRUE)
+  disagree <- sum(call_a != call_b, na.rm = TRUE)
+  if(na.include){
+    # add single NA values
+    disagree <- disagree  + sum(xor(is.na(call_a), is.na(call_b)))
+  } 
+  return(agree/(agree + disagree))
+}
+
+```
+
+
+
+
+# Single cell processing and filtering
+
+The SingleCellExperiment files have already been processed via the `scpca-nf` pipeline, which performs filtering and adds cellhash-based demultiplexing calls using `DropletUtils::HashedDrops` and `Seurat::HTODemux`.
+
+The pipeline also incorporates genetic information inferred from bulk sequencing and performs genetic demultiplexing with [cellsnp](https://cellsnp-lite.readthedocs.io/en/latest/) and [Vireo](https://vireosnp.readthedocs.io/en/latest/).
+
+
+```{r}
+sce <- readRDS(filtered_file)
+```
+
+## Sample information
+
+The metadata in the processed SingleCellExperiment includes information about the samples included in this library:
+
+```{r}
+metadata(sce)$sample_id
+```
+
+
+# Demultiplexing results
+
+The confident calls (by default measures) are stored in the `colData` data frame, prefixed by method, so we will extract those calls for comparisons.
+Doublets here are coded as part of the `NA` group.
+
+```{r}
+demux_calls <- data.frame(
+  hashedDrops = sce$hashedDrops_sampleid,
+  HTODemux = sce$HTODemux_sampleid,
+  vireo = sce$vireo_sampleid
+)
+
+# Summarize all of the calls
+demux_calls |>
+  dplyr::mutate_all(as.factor) |>
+  summary()
+```
+
+In general, we see that `DropletUtils::hashedDrops` tends to make many fewer calls than `Seurat::HTODemux`. 
+In either case, we have a large excess of `NA` calls, where the cellhash-based demultiplexing algorithms did not find a single best call. 
+
+This may partly be a function of the quality of the cellhash results for this particular library.
+As seen in `14-cellhash-exploration.Rmd`, the cellhash data for these samples seems to have a lot of background signal that may make accurate demultiplexing based on cellhash data alone difficult.
+
+The genetic demultiplexing calls made by `vireo` seem to identify the most cells, but the relative frequencies of the samples do not necessarily agree with the results from `HTODemux`.
+This could be because different sets of cells are being called by the two methods, or because the actual calls differ, so we can look in a bit more detail.
+
+## Comparing classifications
+
+To compare the calls between methods, we will build some contingency tables among the call sets. 
```

Comment:
Can you add a sentence here that just briefly explains the comparisons that you are making below? I think it would be helpful since you have the function separately to just explicitly state that you are making two comparisons across all the methods, the first is looking at whether or not the two methods agree on how they call cells without considering cells that did not get a call or were not successfully demultiplexed for that method, and the second with considering all cells including those that were not successfully demultiplexed.